runtime/lua/vim/treesitter/query.lua

```diff
@@ -353,6 +353,12 @@ function Query:iter_captures(node, source, start, stop)
   if type(source) == "number" and source == 0 then
     source = vim.api.nvim_get_current_buf()
   end
+
+  if start == nil and stop == nil then
```

Comment:
Since these two chunks are identical it would be nice to keep it DRY and move this to a function.

Reply:
Good idea, together with *idea 1* this would be more clean.